agent/statsd_server_test.go

```diff
@@ -160,3 +133,150 @@ func TestComposeSetPoints(t *testing.T) {
 		assert.Equal(t, point.Tags, tags)
 	}
 }
+
+func TestPreparePoints(t *testing.T) {
+	now := time.Now().UnixNano()
+	metrics := FixtureMetricMap(now)
+
+	points := prepareMetrics(now, &metrics)
+	assert.Equal(t, len(points), 14)
+	for _, point := range points {
+		assert.Contains(t, point.Name, "test")
+		assert.Equal(t, point.Timestamp, now)
+	}
+}
+
+func TestReceiveMetrics(t *testing.T) {
+	assert := assert.New(t)
+
+	cfg := FixtureConfig()
+	ta := NewAgent(cfg)
+
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	go ta.statsdServer.Run(ctx)
+
+	udpClient, err := net.Dial("udp", ta.statsdServer.MetricsAddr)
```

Comment:
I wonder what happens if the server doesn't start before we try to open a UDP conn here? Seems like it could lead to a flaky test.

I'm a little uncertain about the intent of the test as well. It seems that all it does is fire UDP packets at statsd, which presumably works. Maybe we don't need it.

Reply:
What kind of solution might you suggest? I think we should establish some kind of reasonable time window for the udp server to start. If it does not start within that window, thus failing the test, it would indicate we should file an issue with the dependency.

So right now, the intent of the test is to ensure we can write a metric map to the statsd server with no error. Since the feature is not complete (blocked by #1247) we cannot yet verify the metrics received are written to an event (the `TODO` portion of the test). I still think its important to validate that we're able to establish a udp connection with the statsd server and write metrics to it, otherwise there is something wrong with the semi-complete portion of the feature.

Reply:
We do something similar here to test the udp and tcp sockets: https://github.com/sensu/sensu-go/blob/master/agent/sockets_test.go#L29-L32. But yeah, I agree it's definitely not a unit test, and if kept as is, should definitely be tagged as integration.

I'd like to keep the existing test, and explore implementing a mock as well. However, I'm curious with how we would want to go about the mock. It sounds like we wouldn't be mocking the actual server, correct (`server.Run()`)? Are you seeking a test that would just validate different kinds of messages that could be sent to the server?